ansible/roles/csm.ncn.move_first_master/tasks/main.yml

```diff
@@ -0,0 +1,115 @@
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# (MIT License)
+
+---
+- include_role:
+    name: csm.ncn.download_utils
+- include_role:
+    name: csm.ncn.standardize_hostnames
+
+- name: Get BSS Global cloud-init payload
+  delegate_to: localhost
+  uri:
+    url: "{{ bss_url }}/boot/v1/bootparameters"
+    body_format: json
+    body:
+      hosts:
+        - Global
+    headers:
+      Authorization: "{{ authorization_header }}"
+  register: global_payload
+
+- name: Get first master hostname
+  set_fact:
+    first_master_hostname: "{{ global_payload.json | first | json_query(json_query) }}"
+  vars:
+    json_query: '"cloud-init"."meta-data"."first-master-hostname"'
+
+- name: Pick new first master
+  when:
+    - first_master_hostname == target_hostname
+  delegate_to: localhost
+  set_fact:
+    new_first_master_hostname: "{{ item }}"
+  with_random_choice: "{{ master_ncns | reject('search', target_hostname) | list }}"
+
+- name: Get certificate key
+  when:
+    - first_master_hostname == target_hostname
+  delegate_to: "{{ new_first_master_hostname }}"
+  shell: kubeadm init phase upload-certs --upload-certs 2>&1 | tail -1
+  register: certificate_key
+
+- name: Build kubeadm config builder script
+  when:
+    - first_master_hostname == target_hostname
+  delegate_to: "{{ new_first_master_hostname }}"
+  template:
+    src: "templates/build-kubeadm.sh.j2"
+    dest: "/tmp/build-kubeadm.sh"
+    mode: 0755
+
+- name: Generate kubeadm config using script
+  when:
+    - first_master_hostname == target_hostname
+  delegate_to: "{{ new_first_master_hostname }}"
+  shell: /tmp/build-kubeadm.sh
+
+- name: Copy token certs refresh script to new first master
+  when:
+    - first_master_hostname == target_hostname
+  delegate_to: "{{ new_first_master_hostname }}"
+  copy:
+    src: "files/token-certs-refresh.sh"
+    dest: "/srv/cray/scripts/kubernetes/"
+    mode: 0755
+
+- name: Run token certs refresh script
+  when:
+    - first_master_hostname == target_hostname
+  delegate_to: "{{ new_first_master_hostname }}"
+  shell: /srv/cray/scripts/kubernetes/token-certs-refresh.sh
+
+- name: Setup cron to run refresh script
+  when:
+    - first_master_hostname == target_hostname
+  delegate_to: "{{ new_first_master_hostname }}"
+  cron:
+    name: cray-k8s-token-certs-refresh
+    special_time: hourly
+    user: root
+    job: "/srv/cray/scripts/kubernetes/token-certs-refresh.sh >> /var/log/cray/cron.log 2>&1"
+    cron_file: cray-k8s-token-certs-refresh
+
+- name: Set new first master in BSS global cloud-init metadata
+  when:
+    - first_master_hostname == target_hostname
+  delegate_to: localhost
+  environment:
+    SLS_BASE_URL: "{{ sls_url }}"
+    BSS_BASE_URL: "{{ bss_url }}"
+    HSM_BASE_URL: "{{ hsm_url }}"
+    TOKEN: "{{ token }}"
+  command: csi handoff bss-update-cloud-init --set
+    meta-data.first-master-hostname="{{ new_first_master_hostname }}" --limit Global
+  register: is_member_output
+  ignore_errors: true
```

Comment:
Is this ignorable only because it's manipulating non-critical metadata?

Reply:
No, that's a mistake, an error here would not be good. Good catch.